test/Hpack/DependencySpec.hs

```diff
@@ -109,8 +109,34 @@ spec = do
 
         it "rejects invalid values" $ do
           [i|
-            hpack: 23
-          |] `parsesAs` Left "Error in $.hpack: expected Null, Object, or String, encountered Number"
+            hpack: []
+          |] `parsesAs` Left "Error in $.hpack: expected Null, Object, Number, or String, encountered Array"
+
+        context "when the constraint is a Number" $ do
+          it "accepts 1" $ do
+            [i|
+              hpack: 1
+            |] `parsesAs` Right [("hpack", VersionRange "==1")]
+
+          it "accepts 1.0" $ do
+            [i|
+              hpack: 1.0
+            |] `parsesAs` Right [("hpack", VersionRange "==1.0")]
+
+          it "accepts 0.11" $ do
+            [i|
+              hpack: 0.11
+            |] `parsesAs` Right [("hpack", VersionRange "==0.11")]
+
+          it "accepts 0.110" $ do
+            [i|
+              hpack: 0.110
+            |] `parsesAs` Right [("hpack", VersionRange "==0.110")]
+
+          it "accepts 1e2" $ do
+            [i|
+              hpack: 1e2
+            |] `parsesAs` Right [("hpack", VersionRange "==100")]
```

Comment:
@tfausak we could choose to reject `1e2` as the difference between `1e2` and `100` is observable.  However, the difference between `0.01` and `1e-2` is not observable.

Reply:
I think it's fine to allow `1e2`. It's not a valid version number, so there's no potential for confusion. It's kind of weird, but 🤷‍♂️

Reply:
Agree.